### PR TITLE
Improve feature gate rendering

### PR DESCRIPTION
**Pull request by sftim**
- **Render feature gate descriptions automatically**
  Build a sorted list of feature gates based on the individual files that describe them.
- **Set a title for feature gates in the table**
  If you hover over a feature gate name in the table of feature gates, you (now) see the text description. At least in typical web browsers.

_Previews_
[Before](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates) | [After](https://deploy-preview-44554--kubernetes-io-main-staging.netlify.app/docs/reference/command-line-tools-reference/feature-gates/#feature-gates) (current feature gates)
[Before](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/#descriptions-for-removed-feature-gates) | [After](https://deploy-preview-44554--kubernetes-io-main-staging.netlify.app/docs/reference/command-line-tools-reference/feature-gates-removed/#descriptions-for-removed-feature-gates) (removed feature gates)

/area web-development

### CODE REVIEW

**Comment by sftim** [comment]
@dipesh-rawat would you like to review this?

**Comment by sftim** [comment]
Because the existing file was manually managed, there are some changes. The data driven approach has no mismatches between the tables and the list of descriptions.

**Review by dipesh-rawat — COMMENTED** [review]
Oh no, you've outpaced me! I had the same plan to automate the list with similar changes. Great minds think alike😄!

File: content/en/docs/reference/command-line-tools-reference/feature-gates-removed/index.md [hunk 1]
```diff
@@ -31,159 +31,4 @@ In the following table:
...
-- {{< feature-gate-description name="VolumeScheduling" >}}
-- {{< feature-gate-description name="VolumeSnapshotDataSource" >}}
-- {{< feature-gate-description name="VolumeSubpath" >}}
-- {{< feature-gate-description name="VolumeSubpathEnvExpansion" >}}
-- {{< feature-gate-description name="WarningHeaders" >}}
-- {{< feature-gate-description name="WindowsEndpointSliceProxying" >}}
-- {{< feature-gate-description name="WindowsGMSA" >}}
-- {{< feature-gate-description name="WindowsHostProcessContainers" >}}
-- {{< feature-gate-description name="WindowsRunAsUserName" >}}
+{{< feature-gate-list show-removed="true" >}}
```
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates-removed/index.md:34]: A HTML guidance comment on how to address any issue(s) related to the description items listed below would be helpful here.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/index.md [hunk 2]
```diff
@@ -102,171 +102,9 @@ A *General Availability* (GA) feature is also referred to as a *stable* feature.
...
-- {{< feature-gate-description name="UserNamespacesSupport" >}}
-- {{< feature-gate-description name="ValidatingAdmissionPolicy" >}}
-- {{< feature-gate-description name="VolumeAttributesClass" >}}
-- {{< feature-gate-description name="VolumeCapacityPriority" >}}
-- {{< feature-gate-description name="WatchBookmark" >}}
-- {{< feature-gate-description name="WatchList" >}}
-- {{< feature-gate-description name="WindowsHostNetwork" >}}
-- {{< feature-gate-description name="WinDSR" >}}
-- {{< feature-gate-description name="WinOverlay" >}}
-- {{< feature-gate-description name="ZeroLimitedNominalConcurrencyShares" >}}
```
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/index.md:269]: This change also reminds me that we may want to revise our [existing guidance](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to align with this change.

> Also make sure to move the relevant list entry and [feature-gate-description shortcode](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#feature-gate-description) into the removed feature gates page. The lists are sorted alphabetically.

File: layouts/shortcodes/feature-gate-list.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+<!-- Extract value for the 'include' parameter passed to the shortcode  -->
+{{- $includeValues := .Get "include" -}}
+
+<!-- Sort Feature gate pages list -->
+{{- $sortedFeatureGates := sort ($featureDataFiles.Resources.ByType "page") -}}
+
+<ul class="feature-gate-list">
+  {{- range $featureGateFile := $sortedFeatureGates -}}
+
+    {{- $featureRowCount := 0 -}}
```
Comment (dipesh-rawat) [inline layouts/shortcodes/feature-gate-list.html:15]: This appears to be an unused variable.

File: layouts/shortcodes/feature-gate-table.html [hunk 4]
```diff
@@ -37,6 +37,8 @@
     <tbody>
     {{- range $featureGateFile := $sortedFeatureGates -}}
 
+        {{- $featureRowCount := 0 -}}
```
Comment (dipesh-rawat) [inline layouts/shortcodes/feature-gate-table.html:40]: This appears to be an unused variable.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/index.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/index.md:269]: Ah, good point.

File: content/en/docs/reference/command-line-tools-reference/feature-gates-removed/index.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/reference/command-line-tools-reference/feature-gates-removed/index.md:35]: How's this?

File: layouts/shortcodes/feature-gate-list.html (hunk 3 above)
Comment (sftim) [reply layouts/shortcodes/feature-gate-list.html:15]: Removed. Oops.

**Review by dipesh-rawat — COMMENTED** [review]
Looks good to me, these improvements are impressive. 
I'm already witnessing the advantages of this automation. While reviewing and validating these changes, I noticed in the current website a few feature gate descriptions (e.g: GRPCContainerProbe, CronJobTimeZone) lingering on the [Feature Gates page](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/), even though the feature gate has been moved to the [Feature Gates (removed) page](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/). The output from the shortcode ensure everything displays correctly in the deploy preview. This will prevent similar issues in the future and greatly assist contributors in documenting their features for upcoming release! Great work! 🚀

/lgtm

`/approve` _(Officially not an approver, but unofficially giving the nod of approval_)

**Comment by reylejano** [comment]
This is great work. Thank you for `this`
Contributors working on feature gates don't have to update the new-features or the feature-gates-removed files
/approve

**Pull request merged**